samples/Kconfig

```diff
@@ -90,3 +90,6 @@ config LOG_DEFAULT_LEVEL
 	default 1 if NCS_SAMPLES_DEFAULTS
 
 endif # LOG
+
+config NCS_IS_VARIANT_IMAGE
+	bool "Image is a variant build of another image [READ ONLY]"
```

Comment:
I know the intention is to have this information available in source code.

I assume the number of cases where source code itself need to be aware of this are very few.
So maybe we should handle those case specifically.

Not so fond of user-accessible setting denoted `READ ONLY`.
Alternative, we can look into the possibility of making this promptless, and handled differently.

Reply:
I can't think of a better mechanism than Kconfig.

Adjusting how the option is declared might make sense though.

Reply:
I don't see how we can modify the option if we make it promptless. Wouldn't we have to add some kconfig source code to the build?

Reply:
> I can't think of a better mechanism than Kconfig.

Simply trying to understand exactly the intention of this.
If going to be used in code or similar, Kconfig is best way, if this is purely a build system to build system knowledge, then pure CMake variable could be an option.

I see a risk in having a very generic symbol with a prompt for the following reasons:
- Users are not supposed to change this variable, it may mess up a build
- It is in no way clear what enabling this setting trigger of changes.
  If there were two symbols, like `BOOTLOADER` used in ordinary build `BOOTLOADER_VARIANT`, then I would know that something related to bootloader is affected.
  Here i'm absolutely clueless as what gets adjusted, I would need to search.
  @hakonfam thanks for pointing me to: https://github.com/nrfconnect/sdk-zephyr/blob/93c6a102f28fd00321c8425ddda66d7105ec55c3/include/arch/arm/aarch32/cortex_m/scripts/linker.ld#L35-L39

Now, making a variant is a generic term, and it's true that for now every time we build a variant it will be for slot 1, so in that case it could be fine, but we could also consider if it would be better to have a specific setting to indicate what setting the variant adjusts.

If a variant build in future needs the slot, but also other settings in addition, then all those could be selected by a higher level symbol. I believe that will make it easier to understand what changes a variant build imposes.

Reply:
an example of how we may provide variant specific Kconfig options, not generally available can be seen here:
https://github.com/hakonfam/fw-nrfconnect-nrf-1/pull/12